gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/AbstractDiskRegion.java

```diff
@@ -775,7 +777,15 @@ public int getRecoveredEntryCount() {
   public void incRecoveredEntryCount() {
     this.recoveredEntryCount++;
   }
-  
+
+  public int getInvalidOrTombstoneEntryCount() {
+    return this.invalidOrTombstoneCount;
+  }
+
+  public void incInvalidOrTombstoneEntryCount() {
+    this.invalidOrTombstoneCount++;
+  }
+
```

Comment:
Will this not require a lock or be an AtomicInteger? Same for recoveredEntryCount. I think we can make both as AtomicInteger to be on the safe side. Either way since this is a public API, it should be either clearly marked as "NOT THREAD-SAFE" then all callers ensured to be so, or better will be to make it atomic and avoid the trouble as mentioned.